lib/native/eventstream.ts

```diff
@@ -91,13 +93,21 @@ export class Header {
     private constructor(public name: string, public type: HeaderType, private value?: any) {
     }
 
+    private static isHeaderNameValid(name: string) {
+        return name.length > 0 && name.length <= AWS_MAXIMUM_EVENT_STREAM_HEADER_NAME_LENGTH;
+    }
+
     /**
      * Create a new boolean-valued message header
      *
      * @param name name of the header
      * @param value value of the header
      */
     static newBoolean(name: string, value: boolean): Header {
+        if (!Header.isHeaderNameValid(name)) {
+            throw new CrtError(`Event stream header name (${name}) is not valid`);
```

Comment:
Trivial: The CrtError is identical across all these checks, could this be thrown in the `isHeaderNameValid()` function?

Reply:
Could change it to validateHeaderName() I suppose.